packages/hash/frontend/sentry.client.config.ts

```diff
@@ -2,19 +2,36 @@
 // https://docs.sentry.io/platforms/javascript/guides/nextjs/
 
 import * as Sentry from "@sentry/nextjs";
+import { Replay } from "@sentry/replay";
+import {
+  SENTRY_DSN,
+  SENTRY_REPLAYS_SAMPLING_RATE,
+  VERCEL_ENV,
+} from "./src/lib/public-env";
 
-const dsn = process.env.NEXT_PUBLIC_SENTRY_DSN;
+const replaysSamplingRate = Number.parseInt(
+  SENTRY_REPLAYS_SAMPLING_RATE ?? "0",
+  10,
+);
 
-export const SENTRY_CONFIG = {
-  dsn,
-  enabled: !!dsn,
-  environment:
-    process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_REF === "main"
-      ? "production"
-      : "development",
+Sentry.init({
+  dsn: SENTRY_DSN,
+  enabled: !!SENTRY_DSN,
+  environment: VERCEL_ENV,
+  integrations:
+    replaysSamplingRate > 0 &&
+    replaysSamplingRate <= 1 &&
```

Comment:
It's nice to have guards on the bounds, and makes things self-documenting too. 

But in practice with this integration you can pass a value that is higher or lower than `0.0` or `1.0`. So it's possible to get away with as little as this if you wanted:
```
  environment: VERCEL_ENV,
  integrations:
    replaysSamplingRate
      ? [new Replay(...)]
      : [],
  tracesSampleRate: 1.0,
```

Reply:
Makes sense, addressed in https://github.com/hashintel/hash/pull/1232/commits/aba0eb9b4aee1d43f2948fa4aa42f37f2e1c4ff9 🙂